addons/l10n_it_edi/models/account_edi_proxy_user.py

```diff
@@ -25,3 +30,202 @@ def _compute_proxy_type(self):
+        * transactionId: The fatturapa ID of this request.
+        * error:         An eventual error.
+        * error_level:   Info, warning, error.
+        '''
+        if not files:
+            return {}
+        if self.edi_mode == 'demo':
+            return {file_data['filename']: {'id_transaction': 'demo'} for file_data in files}
+
+        ERRORS = {'EI01': _('Attached file is empty'),
+                  'EI02': _('Service momentarily unavailable'),
+                  'EI03': _('Unauthorized user')}
+
+        server_url = self._get_server_url()
+        results = self._make_request(
+            f'{server_url}/api/l10n_it_edi/1/out/SdiRiceviFile',
+            params={'files': files})
+
+        for filename, vals in results.items():
+            if 'error' in vals:
+                results[filename]['error'] = ERRORS.get(vals.get('error'), _("Unknown error"))
+
+        return results
+
+    def cron_l10n_it_edi_download_and_update(self):
+        """ Crons run with sudo(), with empty recordset. Remember that. """
+        retrigger = False
+        for proxy_user in self.search([('proxy_type', '=', 'l10n_it_edi')]):
+            user = proxy_user.with_company(proxy_user.company_id)
+            if user.edi_mode != 'demo':
+                user._l10n_it_edi_update_send_state()
+                retrigger = retrigger or user._l10n_it_edi_download()
+            else:
+                user._l10n_it_edi_demo_mode_update()
+
+        # Retrigger download if there are still some on the server
+        if retrigger:
+            _logger.info('Retriggering "Receive invoices from the SdI"...')
+            self.env.ref('l10n_it_edi.ir_cron_l10n_it_edi_download_and_update')._trigger()
+
+    def _l10n_it_edi_download(self):
+        """ Check the proxy for incoming invoices for a specified proxy user.
+            :param transaction_ids: id of the SdI transaction for communication with the IAP proxy.
+        """
+        server_url = self._get_server_url()
+
+        # Download invoices
+        invoices_data = {}
+        try:
+            invoices_data = self._make_request(f'{server_url}/api/l10n_it_edi/1/in/RicezioneInvoice',
+                params={'recipient_codice_fiscale': self.company_id.l10n_it_codice_fiscale})
+        except AccountEdiProxyError as e:
+            _logger.error('Error while receiving invoices from the SdI: %s', e)
+            return False
+
+        # Process the downloaded invoices
+        processed = self._l10n_it_edi_process_downloads(invoices_data)
+        if processed['proxy_acks']:
+            try:
+                self._make_request(
+                    f'{server_url}/api/l10n_it_edi/1/ack',
+                    params={'transaction_ids': processed['proxy_acks']})
+            except AccountEdiProxyError as e:
+                _logger.error('Error while receiving file from the SdI: %s', e)
+
+        return processed['retrigger']
+
+    def _l10n_it_edi_store_download(self, filename, content, key):
```

Comment:
Such name is missleading. On `account.move`, having `_l10n_it_edi_import_invoice` is more accurate

Reply:
I've called it `create_move_with_attachment` as that's what it does.